File: backend/src/main/java/com/codesqaude/cocomarco/domain/issue/Issue.java
package com.codesqaude.cocomarco.domain.issue;

import com.codesqaude.cocomarco.domain.milestone.Milestone;
import lombok.Getter;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
public class Issue {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String text;
    private LocalDateTime writingTime;

    @Enumerated(EnumType.STRING)
    private IssueStatus status;

    @OneToMany(mappedBy = "issue")
    private List<Comment> comments = new ArrayList<>();

    @OneToMany(mappedBy = "issue")
    private List<Assignment> assignments = new ArrayList<>();

    @OneToMany(mappedBy = "label")
    private List<IssueLabel> issueLabels = new ArrayList<>();

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "milestone_id")
    private Milestone milestone;
}

File: iOS/Issue-Trackker/Issue-Trackker/Model/Data/Entity/Issue.swift
//
//  Issue.swift
//  Issue-Trackker
//
//  Created by 심영민 on 2021/06/13.
//

import Foundation

struct Issue: Decodable {
    var id: Int
    var title: String
    var writer: String
    var writeTime: String
    var milestone: Milestone
    var assignments: [Assignment]
    var labels: [Label]
}



